withTransaction.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
```

Comment:
We don't want to add this file do we?

Reply:
addressed in [405aed4](https://github.com/mongodb/mongo-python-driver/pull/2600/commits/405aed4bbffd16bfe9102c6096b3819b1303561f) and [ef9d85c](https://github.com/mongodb/mongo-python-driver/pull/2600/commits/ef9d85c5fd5ae9a914805b8b0710671d23601ed4)
resolving conversation :)